pkg/supervisor/supervisor.go

```diff
@@ -357,7 +357,7 @@ func (s *Supervisor) shouldKillProcess(ph procHandle) (bool, error) {
 // - inject k0s embedded bins into path
 func getEnv(dataDir, component string, keepEnvPrefix bool) []string {
```

Comment:
This function seems strange.

- The `i` is not incremented when an override is encountered. Doesn't this make the `env[i]=x` later modify the wrong entry?
- The `k0sManaged` is added even if it already exists. 
- The `i++` + `return env[:i]` is kind of suspicious. 

Even if it works, in the sake of clarity, I think this should be simplified.

Reply:
I don't understand what it does in general.

Any `COMPONENT_XXX` encountered will be converted to `XXX` once. After that any `COMPONENT_YYY` will be skipped?

Reply:
Hmm, no. It will skip any following `XXX` after `COMPONENT_XXX` has been encountered.

Reply:
So, `COMPONENT_HTTP_PROXY=abcd` will always become `HTTP_PROXY` and any following `HTTP_PROXY` will be skipped.

`COMPONENT_FOO=z` will go through as-is unless `KeepEnvPrefix` is false, in which case it will become `FOO=z` and any following `FOO=x` will be skipped.

Reply:
Ok, so maybe it does what it is supposed to. Maybe there isn't a one-off bug.

But it is very weird.

Reply:
> Ok, so maybe it does what it is supposed to. Maybe there isn't a one-off bug.

There are tests for it, so I think the implementation should be "correct".

> But it is very weird.

It _is_ weird :see_no_evil: